### PR TITLE
Duplicate vote fix

### PR DESCRIPTION
**Pull request by SargeKhan**

### CODE REVIEW

File: logic/vote.js [hunk 1]
```diff
@@ -95,6 +95,12 @@ Vote.prototype.verify = function (trs, sender, cb) {
 		return setImmediate(cb, 'Voting limit exceeded. Maximum is 33 votes per transaction');
 	}
 
+	for (var i = 0; i < trs.asset.votes.length; i++) {
+		if (trs.asset.votes.filter(function (v) { return v === trs.asset.votes[i]; }).length > 1) {
+			return setImmediate(cb, 'Duplicate votes are not allowed');
+		}
+	}
```
Comment (MaciejBaj) [inline logic/vote.js]: could be also done with lodash `uniq`
```
if ( trs.asset.votes.length !== _.uniq(trs.asset.votes).length) {
   return setImmediate(cb, 'Duplicate votes are not allowed');
}
```

Comment (SargeKhan) [inline logic/vote.js]: Didn't want to import lodash in the file for use in only one condition. But non-nested if condition looks better. So, I'll update it. :+1: Thanks

Comment (Isabello) [inline logic/vote.js]: I appreciate that @SargeKhan :)

File: test/api/delegates.js [hunk 2]
```diff
@@ -102,40 +102,45 @@ describe('PUT /api/accounts/delegates with funds', function () {
...
 	it('when upvoting and downvoting within same request should fail', function (done) {
-		var votedDelegate = '"+' + node.eAccount.publicKey + '","-' + node.eAccount.publicKey + '"';
+		var votedDelegate = ['-' + node.eAccount.publicKey, '+' + node.eAccount.publicKey];
 
 		putAccountsDelegates({
 			secret: account.password,
-			delegates: [votedDelegate]
+			delegates: votedDelegate
 		}, function (err, res) {
+			var expectedErrors = [
```
Comment (4miners) [inline test/api/delegates.js]: Those error we're check against not looks like related to upvote/downvote in same transaction. We should explicitly not allow that, with proper error message.

File: logic/vote.js [hunk 3]
```diff
@@ -95,6 +96,10 @@ Vote.prototype.verify = function (trs, sender, cb) {
 		return setImmediate(cb, 'Voting limit exceeded. Maximum is 33 votes per transaction');
 	}
 
+	if (trs.asset.votes.length > _.uniq(trs.asset.votes).length) {
```
Comment (4miners) [inline logic/vote.js]: We should compare public keys instead of votes, that will cover unvote/downvote in same transaction.

Comment (SargeKhan) [inline logic/vote.js]: that's great advice

**Pull request merged**